bug: integration test test_detect_calculator_has_uia fails — Calculator only exposes vision method

## Problem
The integration test `test_detect_calculator_has_uia` in `tests/integration/test_unified_app_model.py:189` fails because Calculator only returns `['vision']` as available detection methods, missing `uia`.

## Steps to Reproduce
1. Run: `python -m pytest tests/integration/test_unified_app_model.py::TestDetectionChainCalculator::test_detect_calculator_has_uia -v`
2. Test asserts `"uia" in method_names` but actual methods are `['vision']`

## Expected
Calculator (UWP app) should support UIA detection method in addition to vision.

## Actual
```
AssertionError: Calculator should support UIA, got methods: ['vision']
assert 'uia' in ['vision']
```

## Context
- Related to #785 (Calculator UWP invisible to list apps/see)
- This blocks CI from passing on the `desktop` marker tests
- Commit: 42443f1

## Environment
- Windows 11 10.0.26200.8037
- naturo 0.3.1
- Runner: ROBOT-COMPILE

**[QA-Mariana]**

--- a/naturo/detect/probes.py
+++ b/naturo/detect/probes.py
@@ -594,15 +594,56 @@ def probe_uia(pid: int, exe: str, hwnd: Optional[int] = None) -> Optional[Intera
 
         element = uia.ElementFromHandle(target_hwnd)
         if element:
-            capabilities = ["click", "type", "find", "tree", "screenshot"]
+            # Verify the element has meaningful children (not an empty shell).
+            # WinUI 3 apps may return an element for the main window that has
+            # no useful UIA subtree.
+            try:
+                child_count = element.CurrentChildren and element.CurrentChildren.Length
+            except Exception:
+                child_count = None
 
-            return InteractionMethod(
-                method=InteractionMethodType.UIA,
-                priority=METHOD_PRIORITY[InteractionMethodType.UIA],
-                status=ProbeStatus.AVAILABLE,
-                capabilities=capabilities,
-                confidence=0.9,
-            )
+            if child_count:
+                capabilities = ["click", "type", "find", "tree", "screenshot"]
+                return InteractionMethod(
+                    method=InteractionMethodType.UIA,
+                    priority=METHOD_PRIORITY[InteractionMethodType.UIA],
+                    status=ProbeStatus.AVAILABLE,
+                    capabilities=capabilities,
+                    confidence=0.9,
+                )
+
+            # (#841) Mirror Strategy 1: probe AFH and WinUI child windows.
+            for child_hwnd in _find_afh_content_children(target_hwnd):
+                child_el = uia.ElementFromHandle(child_hwnd)
+                if child_el:
+                    logger.debug(
+                        "comtypes UIA probe succeeded via AFH child HWND %s",
+                        child_hwnd,
+                    )
+                    capabilities = ["click", "type", "find", "tree", "screenshot"]
+                    return InteractionMethod(
+                        method=InteractionMethodType.UIA,
+                        priority=METHOD_PRIORITY[InteractionMethodType.UIA],
+                        status=ProbeStatus.AVAILABLE,
+                        capabilities=capabilities,
+                        confidence=0.9,
+                    )
+
+            for child_hwnd in _find_winui_content_children(target_hwnd):
+                child_el = uia.ElementFromHandle(child_hwnd)
+                if child_el:
+                    logger.debug(
+                        "comtypes UIA probe succeeded via WinUI child HWND %s",
+                        child_hwnd,
+                    )
+                    capabilities = ["click", "type", "find", "tree", "screenshot"]
+                    return InteractionMethod(
+                        method=InteractionMethodType.UIA,
+                        priority=METHOD_PRIORITY[InteractionMethodType.UIA],
+                        status=ProbeStatus.AVAILABLE,
+                        capabilities=capabilities,
+                        confidence=0.9,
+                    )
 
     except ImportError:
         logger.debug("comtypes not available for UIA probe — install with: pip install comtypes")

--- a/naturo/detect/probes.py
+++ b/naturo/detect/probes.py
@@ -483,6 +483,32 @@ def _find_winui_content_children(parent_hwnd: int) -> List[int]:
         return []
 
 
+def _comtypes_element_is_useful(element: object, uia: object) -> bool:
+    """Check whether a comtypes UIA element has actionable children.
+
+    For WinUI 3 apps the top-level window element exists but contains no
+    buttons, text fields, or other interactive controls.  We check via
+    FindAll with TreeScope_Children and a TrueCondition.
+
+    Args:
+        element: IUIAutomationElement from comtypes.
+        uia: IUIAutomation instance.
+
+    Returns:
+        True if the element has at least one child (useful for interaction).
+    """
+    try:
+        # TreeScope_Children = 2, CreateTrueCondition returns match-all
+        condition = uia.CreateTrueCondition()  # type: ignore[union-attr]
+        children = element.FindAll(2, condition)  # type: ignore[union-attr]
+        count = getattr(children, "Length", 0)
+        return count > 0
+    except Exception:
+        # If we can't enumerate children, assume it's useful
+        # (conservative — avoid false negatives)
+        return True
+
+
 def probe_uia(pid: int, exe: str, hwnd: Optional[int] = None) -> Optional[InteractionMethod]:
     """Probe for UI Automation availability.
 
@@ -592,18 +618,29 @@ def probe_uia(pid: int, exe: str, hwnd: Optional[int] = None) -> Optional[Intera
             interface=_IUIAutomation,
         )
 
-        element = uia.ElementFromHandle(target_hwnd)
-        if element:
-            # Verify the element has meaningful children (not an empty shell).
-            # WinUI 3 apps may return an element for the main window that has
-            # no useful UIA subtree.
-            try:
-                child_count = element.CurrentChildren and element.CurrentChildren.Length
-            except Exception:
-                child_count = None
+        capabilities = ["click", "type", "find", "tree", "screenshot"]
 
-            if child_count:
-                capabilities = ["click", "type", "find", "tree", "screenshot"]
+        element = uia.ElementFromHandle(target_hwnd)
+        if element and _comtypes_element_is_useful(element, uia):
+            return InteractionMethod(
+                method=InteractionMethodType.UIA,
+                priority=METHOD_PRIORITY[InteractionMethodType.UIA],
+                status=ProbeStatus.AVAILABLE,
+                capabilities=capabilities,
+                confidence=0.9,
+            )
+
+        # (#841) Mirror Strategy 1: probe AFH and WinUI child windows.
+        # Standalone WinUI 3 apps (Calculator, Paint) have a valid top-level
+        # element but no actionable children — the real UI is inside
+        # DesktopWindowXamlSource or CoreWindow child HWNDs.
+        for child_hwnd in _find_afh_content_children(target_hwnd):
+            child_elem = uia.ElementFromHandle(child_hwnd)
+            if child_elem and _comtypes_element_is_useful(child_elem, uia):
+                logger.debug(
+                    "UIA comtypes probe succeeded via AFH child HWND %s",
+                    child_hwnd,
+                )
                 return InteractionMethod(
                     method=InteractionMethodType.UIA,
                     priority=METHOD_PRIORITY[InteractionMethodType.UIA],
@@ -612,38 +649,20 @@ def probe_uia(pid: int, exe: str, hwnd: Optional[int] = None) -> Optional[Intera
                     confidence=0.9,
                 )
 
-            # (#841) Mirror Strategy 1: probe AFH and WinUI child windows.
-            for child_hwnd in _find_afh_content_children(target_hwnd):
-                child_el = uia.ElementFromHandle(child_hwnd)
-                if child_el:
-                    logger.debug(
-                        "comtypes UIA probe succeeded via AFH child HWND %s",
-                        child_hwnd,
-                    )
-                    capabilities = ["click", "type", "find", "tree", "screenshot"]
-                    return InteractionMethod(
-                        method=InteractionMethodType.UIA,
-                        priority=METHOD_PRIORITY[InteractionMethodType.UIA],
-                        status=ProbeStatus.AVAILABLE,
-                        capabilities=capabilities,
-                        confidence=0.9,
-                    )
-
-            for child_hwnd in _find_winui_content_children(target_hwnd):
-                child_el = uia.ElementFromHandle(child_hwnd)
-                if child_el:
-                    logger.debug(
-                        "comtypes UIA probe succeeded via WinUI child HWND %s",
-                        child_hwnd,
-                    )
-                    capabilities = ["click", "type", "find", "tree", "screenshot"]
-                    return InteractionMethod(
-                        method=InteractionMethodType.UIA,
-                        priority=METHOD_PRIORITY[InteractionMethodType.UIA],
-                        status=ProbeStatus.AVAILABLE,
-                        capabilities=capabilities,
-                        confidence=0.9,
-                    )
+        for child_hwnd in _find_winui_content_children(target_hwnd):
+            child_elem = uia.ElementFromHandle(child_hwnd)
+            if child_elem and _comtypes_element_is_useful(child_elem, uia):
+                logger.debug(
+                    "UIA comtypes probe succeeded via WinUI child HWND %s",
+                    child_hwnd,
+                )
+                return InteractionMethod(
+                    method=InteractionMethodType.UIA,
+                    priority=METHOD_PRIORITY[InteractionMethodType.UIA],
+                    status=ProbeStatus.AVAILABLE,
+                    capabilities=capabilities,
+                    confidence=0.9,
+                )
 
     except ImportError:
         logger.debug("comtypes not available for UIA probe — install with: pip install comtypes")

--- a/tests/integration/test_unified_app_model.py
+++ b/tests/integration/test_unified_app_model.py
@@ -176,14 +176,23 @@ class TestDetectionChainCalculator:
 
     def test_detect_calculator_framework(self, calculator_app, detect_chain):
         """Calculator should be detected with some framework info."""
-        result = detect_chain(pid=calculator_app, app_name="Calculator")
+        # (#841) Pass exe= so _find_window_by_process_name can resolve
+        # the window for WinUI 3 Calculator where the launcher PID may
+        # differ from the window-owning process.
+        result = detect_chain(pid=calculator_app, exe="CalculatorApp.exe",
+                              app_name="Calculator")
 
         assert result.pid == calculator_app
         assert len(result.methods) > 0, "Should detect at least one interaction method"
 
     def test_detect_calculator_has_uia(self, calculator_app, detect_chain):
         """Calculator (UWP) should support UIA — the primary method for UWP apps."""
-        result = detect_chain(pid=calculator_app, app_name="Calculator")
+        # (#841) Retry detection — WinUI 3 apps need time for UIA tree
+        # initialization, matching the Notepad pattern (#697).
+        result = _detect_with_retry(
+            detect_chain, pid=calculator_app, exe="CalculatorApp.exe",
+            app_name="Calculator",
+        )
 
         method_names = [m.method.value for m in result.methods]
         assert "uia" in method_names, (
@@ -192,7 +201,11 @@ class TestDetectionChainCalculator:
 
     def test_detect_calculator_best_method(self, calculator_app, detect_chain):
         """Calculator's best method should be UIA."""
-        result = detect_chain(pid=calculator_app, app_name="Calculator")
+        # (#841) Retry detection — UIA tree may need extra init time.
+        result = _detect_with_retry(
+            detect_chain, pid=calculator_app, exe="CalculatorApp.exe",
+            app_name="Calculator",
+        )
 
         best = result.best_method()
         assert best is not None, "Should have a best method"

--- a/tests/integration/test_unified_app_model.py
+++ b/tests/integration/test_unified_app_model.py
@@ -323,8 +323,8 @@ class TestDetectionCacheIntegrity:
 
     def test_cache_per_pid(self, notepad_app, calculator_app, detect_chain):
         """Different PIDs should have separate cache entries."""
-        result_notepad = detect_chain(pid=notepad_app, app_name="Notepad")
-        result_calc = detect_chain(pid=calculator_app, app_name="Calculator")
+        result_notepad = detect_chain(pid=notepad_app, app_name="Notepad", exe="notepad.exe")
+        result_calc = detect_chain(pid=calculator_app, app_name="Calculator", exe="CalculatorApp.exe")
 
         assert result_notepad.pid != result_calc.pid
         assert result_notepad.app_name != result_calc.app_name

--- a/tests/test_winui3_uia_probe_785.py
+++ b/tests/test_winui3_uia_probe_785.py
@@ -117,3 +117,23 @@ class TestComtypesFallbackWinUI:
         assert result is not None
         assert result.method.value == "uia"
         assert mock_core.get_element_tree.call_count == 3
+
+    @patch("naturo.detect.probes.platform")
+    @patch("naturo.detect.probes._find_main_window", return_value=1001)
+    @patch("naturo.detect.probes._find_afh_content_children", return_value=[])
+    @patch("naturo.detect.probes._find_winui_content_children", return_value=[])
+    def test_all_fallbacks_fail_returns_none(
+        self, mock_winui, mock_afh, mock_main, mock_platform,
+    ):
+        """When main, AFH, and WinUI all fail, probe returns None
+        (comtypes not available on Linux)."""
+        mock_platform.system.return_value = "Windows"
+
+        mock_core = MagicMock()
+        mock_core.get_element_tree.return_value = None
+
+        with patch("naturo.detect.probes._get_native_core", return_value=mock_core):
+            result = probe_uia(pid=100, exe="CalculatorApp.exe", hwnd=None)
+
+        # On Linux, comtypes import fails, so result is None
+        assert result is None

--- a/tests/test_winui3_uia_probe_785.py
+++ b/tests/test_winui3_uia_probe_785.py
@@ -85,3 +85,35 @@ class TestProbeUiaWinuiFallback:
 
         assert result is not None
         mock_core.get_element_tree.assert_called_once()
+
+
+class TestComtypesFallbackWinUI:
+    """#841: comtypes fallback must also probe WinUI child windows.
+
+    Actual comtypes testing requires Windows with the COM DLL loaded.
+    These tests verify the Strategy 1 (native DLL) path covers the same
+    child-window probing logic that was added to Strategy 2 (comtypes).
+    The integration test test_detect_calculator_has_uia exercises the
+    full path on a real Windows desktop.
+    """
+
+    @patch("naturo.detect.probes.platform")
+    @patch("naturo.detect.probes._find_main_window", return_value=1001)
+    @patch("naturo.detect.probes._find_afh_content_children", return_value=[])
+    @patch("naturo.detect.probes._find_winui_content_children", return_value=[2001, 2002])
+    def test_winui_fallback_tries_multiple_children(
+        self, mock_winui, mock_afh, mock_main, mock_platform,
+    ):
+        """When multiple WinUI children exist, probe tries each in order."""
+        mock_platform.system.return_value = "Windows"
+
+        mock_core = MagicMock()
+        # Main HWND empty, first WinUI child empty, second succeeds
+        mock_core.get_element_tree.side_effect = [None, None, {"role": "Window"}]
+
+        with patch("naturo.detect.probes._get_native_core", return_value=mock_core):
+            result = probe_uia(pid=100, exe="CalculatorApp.exe", hwnd=None)
+
+        assert result is not None
+        assert result.method.value == "uia"
+        assert mock_core.get_element_tree.call_count == 3

